[AY1920S1-CS2113T-W12-2] Financial Ghost

therealnickcheong opened the pull request:
@chengweixuan
@ChenChao19
@cctt1014
@Chianhaoplanks 
@therealnickcheong

flxffy commented on src/main/java/DataTransfer.java:
```diff
@@ -0,0 +1,94 @@
+public interface DataTransfer {
+
+    int NUMBEROFMONTHS = 3;
+
```
Please refer to the style guide for proper variable naming. Constants should also be static final. https://nuscs2113-ay1920s1.github.io/website/coding-standards/java/intermediate.html#naming

flxffy commented on src/main/java/DataTransfer.java:
```diff
@@ -0,0 +1,94 @@
+        yData.add(account.getTotalIncome());
+        yData.add(account.getTotalExp());
+        return Histogram.getHistogram("The Month Report", xData, yData);
+    }
```
JavaDoc comment doesn't reflect what the method is doing. Otherwise it's neat and concise.

flxffy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,80 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+import controlpanel.*;
```
Imported classes should always be listed explicitly.  https://nuscs2113-ay1920s1.github.io/website/coding-standards/java/intermediate.html#package-and-import-statements

flxffy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,80 @@
+            ui.appendToOutput(ui.showLine());
+        }
+        return new String[]{ui.getOutputString(), ui.getGraphContainerString()};
+    }
```
This method looks like it's doing more than just getting a response.

flxffy commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,191 @@
+                if (input.startsWith("graph finance status /until ")) {
+                    String dateString = input.split(" /until ")[1];
+                    graphContainer.getChildren().addAll(
+                            DataTransfer.getCurrFinance(duke.getAccount(), Parser.shortcutTime(dateString))
```
Indentation for the switch statement! Also, why are commands being parsed here?

flxffy commented on src/main/java/controlpanel/MoneyStorage.java:
```diff
@@ -0,0 +1,285 @@
+                        info[0].equals("G") || info[0].equals("INS") || info[0].equals("INIT") ||
+                        info[0].equals("LOA") || info[0].equals("BAN"))) {
+                    throw new DukeException("OOPS!! Your file has been corrupted/ input file is invalid!");
+                }
```
Think about what your switch statement is doing and how this could fit into it.

flxffy commented on src/main/java/controlpanel/MoneyStorage.java:
```diff
@@ -0,0 +1,285 @@
+                        info[0].equals("LOA") || info[0].equals("BAN"))) {
+                    throw new DukeException("OOPS!! Your file has been corrupted/ input file is invalid!");
+                }
+                switch(info[0]) {
```
The switch statement is very lengthy and can be refactored. Apply SLAP.

flxffy commented on src/main/java/controlpanel/MoneyStorage.java:
```diff
@@ -0,0 +1,285 @@
+        return account;
+    }
+
+    public void writeToFile(Account account) {
```
Apply SLAP.

flxffy commented on src/main/java/controlpanel/Parser.java:
```diff
@@ -0,0 +1,213 @@
+     * @return formatted Date based on user inputted date
+     * @throws ParseException if invalid date is parsed
+     */
+    public static LocalDate shortcutTime(String dateStr) throws ParseException {
```
A lot of repeated code resulting in a length method.

flxffy commented on src/main/java/controlpanel/Ui.java:
```diff
@@ -0,0 +1,76 @@
+                + "     What can I do for you?\n"
+                + "____________________________________________________________\n";
+        return ("Hello from\n" + logo + greeting);
+    }
```
Why is this still duke?

flxffy commented on src/main/java/guicommand/UserIcon.java:
```diff
@@ -0,0 +1,47 @@
+        BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+        FileChooser chooser = new FileChooser();
+        chooser.setTitle("Select a picture:");
+        File defaultDirectory = new File("D:/");
```
Remember that your product needs to be cross-platform compatible, as specified in the project constraints.

flxffy commented on src/main/java/help/MemorisePreviousFunctions.java:
```diff
@@ -0,0 +1,71 @@
+        this.flag = true;
+    }
+
+    public void setFlagTrue() { //the command before this is up
```
I don't understand your comment

flxffy commented on src/main/java/money/Account.java:
```diff
@@ -0,0 +1,263 @@
+        populateCurrentMonthLists();
+//        if (account.isInitialised()) {
+//            toInitialize = false;
+//        } else { toInitialize = true; }
```
Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.

flxffy commented on src/main/java/money/Account.java:
```diff
@@ -0,0 +1,263 @@
+    public void initialize(float userSavings, float avgExp) {
+        LocalDate nowDate = LocalDate.now();
+        Income initialSavings;
+        if (userSavings > avgExp * 6) {
+            this.baseSavings = avgExp * 6;
```
What is `6`? Avoid magic numbers.

flxffy commented on src/main/java/money/Account.java:
```diff
@@ -0,0 +1,263 @@
+                    return g1.getPriority().compareTo(g2.getPriority());
+                }
+            }
+        });
```
Good use of comparators.

flxffy commented on src/main/java/money/Bill.java:
```diff
@@ -0,0 +1,27 @@
+    public String getNextPayDay() { return nextPayDay.format(dateTimeFormatter); }
+
+    public void setNextPayDay() { nextPayDay.plusMonths(1); }
+
```
Refer to the style guide. Methods should not be condensed to single lines, no matter how short.

flxffy commented on src/main/java/money/Goal.java:
```diff
@@ -0,0 +1,61 @@
+        }
+        default: {
+            break;
+        }
```
Good attempt at using enums. Take a look at the documentation for enum and you'll find a more effective way of implementing this :)

flxffy commented on src/main/java/money/Instalment.java:
```diff
@@ -0,0 +1,62 @@
+    private LocalDate endDate;
+    private int numOfPaymentsReq;
+    private float AIR;
+    private float MIR;
```
What are `AIR` and `MIR`? And why are they in uppercase? Are these constants?

flxffy commented on src/main/java/money/Loan.java:
```diff
@@ -0,0 +1,172 @@
+     * @throws ParseException If invalid date is parsed
+     */
+    public void settleLoanDebt(float amount) throws ParseException {
+        if (amount == -2) {
```
What is -2? Magic number

flxffy reviewed (commented):
Please familiarise yourselves with SoC, SRP and SLAP.  On the first look, components look tightly coupled (I may be wrong about this). Methods, especially `execute` tend to be very lengthy. 

Things to note:
1. Use the check style! Your code shows unfamiliarity with the style guide.
2. Remnants of duke that don't seem relevant to your product.
3. Why are your tests commented out?
4. Histogram is well-organised and well done. However you still need to use the check style.
5. Good use of enums, keep this up 👍

ChenChao19 committed:
Slap InstalmentTest into separate Test files

ChenChao19 committed:
Slap, Checkstyle and Javadoc Account

ChenChao19 committed:
Refactor GoalsTest, IncomeTest and LoansTest to under money package.

ChenChao19 committed:
Merge branch 'master' into master

Commit:
Undo for DeleteBankAccount

Commit:
Undo for DeleteBankAccount

ChenChao19 committed:
Merge pull request #129 from Chianhaoplanks/master

Done Undo for Settle Split

ChenChao19 committed:
Merge pull request #131 from ChenChao19/master

Refactor test codes

Commit:
Undo for DeleteBankAccount

Commit:
Edited MoneyStorage:

ChenChao19 committed:
Merge pull request #132 from Chianhaoplanks/master

Undo Command refined Done

ChenChao19 committed:
Chekstyle, Javadoc Autocomplete, history and MainWindow.

ChenChao19 committed:
Merge pull request #133 from ChenChao19/master

Chekstyle, Javadoc Autocomplete, history and MainWindow.

Commit:
Fixed some checkstyle + Deleted Bills

Chianhaoplanks committed:
Merge pull request #134 from Chianhaoplanks/master

Fixed some checkstyle + Deleted Bills

Commit:
Fixed author tags

chengweixuan committed:
Merge pull request #135 from Chianhaoplanks/master

Fixed author tags

ChenChao19 committed:
Change size of mainwindow

ChenChao19 committed:
Merge pull request #136 from ChenChao19/master

Change size of mainwindow

cctt1014 committed:
Merge pull request #25 from AY1920S1-CS2113T-W12-2/master

update the code

cctt1014 committed:
improve ui

cctt1014 committed:
Add Split Pane and Improve UI #137

Add Split Pane and Improve UI

chengweixuan committed:
drew casper and friend

chengweixuan committed:
centred casper

chengweixuan committed:
shifted casper's friend to the right

chengweixuan committed:
centred casper's eyes

Chianhaoplanks committed:
Merge pull request #138 from chengweixuan/master

Added Casper and Friend to UI start screen

cctt1014 committed:
Merge pull request #26 from AY1920S1-CS2113T-W12-2/master

update the code

chengweixuan committed:
updated README

chengweixuan committed:
added updated UG and DG

chengweixuan committed:
added updated UG

chengweixuan committed:
Merge remote-tracking branch 'upstream/master'

chengweixuan committed:
added jar v1.3

chengweixuan committed:
Merge pull request #139 from chengweixuan/master

Updated UG, DG, README with new UI and Jar release v1.3

cctt1014 committed:
add code quality tester

chengweixuan committed:
added JUnit for expenditure

chengweixuan committed:
Merge remote-tracking branch 'upstream/master'

chengweixuan committed:
added jar file

cctt1014 committed:
Merge pull request #27 from AY1920S1-CS2113T-W12-2/master

update the code

chengweixuan committed:
Merge pull request #143 from chengweixuan/master

Added JUnit for expenditure and fixed jar file

chengweixuan committed:
Merge pull request #142 from cctt1014/master

Add code quality tester

chengweixuan committed:
added jar file

chengweixuan committed:
Merge remote-tracking branch 'upstream/master'

chengweixuan committed:
Merge pull request #144 from chengweixuan/master

Added Jar file

cctt1014 committed:
update code quality tester

chengweixuan committed:
finished JUnit for expenditure

chengweixuan committed:
added JUnit for check income command

chengweixuan committed:
added check expenditure Junit

therealnickcheong committed:
fixed init command bug and Handled FileNotFoundException in MoneyStorage moneyAccount.txt now in dataFG folder

therealnickcheong committed:
Merge pull request #145 from therealnickcheong/master

fixed init command bug and Handled FileNotFoundException

cctt1014 committed:
Merge pull request #28 from AY1920S1-CS2113T-W12-2/master

update the code

chengweixuan committed:
Delete README.md

Commit:
Added UndoCommandHandlerTest

Commit:
Fixed IOException in FindTest

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W12-2/main

Commit:
Fixed IOException in IncomeTest

Commit:
Fixed IOException not being thrown for tests that load test-txt

Chianhaoplanks committed:
Merge pull request #146 from Chianhaoplanks/master

Added Test Code for UndoCommandHandler

therealnickcheong committed:
commented out change user icon to enable launchable Jar file

therealnickcheong committed:
cleared most of the issues raised during mock pe

therealnickcheong committed:
Merge pull request #217 from therealnickcheong/master

fixed most of the issues raised in the Mock pe

ChenChao19 committed:
fix bugs

ChenChao19 committed:
Merge remote-tracking branch 'upstream/master'

ChenChao19 committed:
update autocomplete

ChenChao19 committed:
Merge pull request #218 from ChenChao19/master

update autocomplete and write test for help package

therealnickcheong committed:
checkstyle and javaDoc all Goal commands, start and init

therealnickcheong committed:
Merge pull request #219 from therealnickcheong/master

checkstyle and javaDoc all Goal commands, start and init

cctt1014 committed:
Merge pull request #29 from AY1920S1-CS2113T-W12-2/master

update the code

cctt1014 committed:
add exception and test code for BankTracker.java

cctt1014 committed:
Merge pull request #220 from cctt1014/master

Add exception and test code for BankTracker.java

chengweixuan committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	dataFG/moneyAccount.txt

Commit:
commit to merge team repo

Commit:
Merged with team repo

ChenChao19 committed:
update autocomplete

ChenChao19 committed:
Merge pull request #222 from ChenChao19/master

update autocomplete

ChenChao19 committed:
handle exception for add instalment

ChenChao19 committed:
update autocomplete

ChenChao19 committed:
Merge pull request #223 from ChenChao19/master

Exception handling

ChenChao19 committed:
handle deletion accepting negative value exception

ChenChao19 committed:
Merge pull request #224 from ChenChao19/master

handle deletion accepting negative value exception

chengweixuan committed:
fixed decimal places for all toStrings

chengweixuan committed:
Fixed exceptions for delete expenditure and income

chengweixuan committed:
fixed exceptions for archive commands

chengweixuan committed:
fixed exceptions for list loans and all loan command. Fixed dp issue for loans

chengweixuan committed:
settled exceptions for settle loan and delete loan command

chengweixuan committed:
Finished JUnit for Archive Commands

chengweixuan committed:
fixed JUnit for Income and Expenditure

chengweixuan committed:
updated JUnit for loans, expenditure and income

chengweixuan committed:
removed split expenditure from code

chengweixuan committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	dataFG/moneyAccount.txt

chengweixuan committed:
Merge pull request #225 from chengweixuan/master

Cleaned up code for Archve. Added JUnit for Archive. Cleaned up JUnit. Removed split expenditure function (buggy)

chengweixuan committed:
Completed JUnit for TimingShortCuts

chengweixuan committed:
Checkstyled parser and Storage

chengweixuan committed:
Checkstyle for Income, Expenditure, Loans and Archive commands

chengweixuan committed:
Merge remote-tracking branch 'upstream/master'

chengweixuan committed:
Merge pull request #226 from chengweixuan/master

Added JUnit for TimingShortcuts and Checkstyle for Income, Expenditure, Archive and Loan commands. Checkstyle for Parser completed

chengweixuan committed:
Slapped SettleLoans and updated AboutUs

therealnickcheong committed:
added 2dp and fixed negative index input

therealnickcheong committed:
Merge pull request #227 from therealnickcheong/master

added 2dp and fixed negative index input

cctt1014 committed:
change a graph command format in autocomplete

cctt1014 committed:
add exception to show errors in wrong graph command format

cctt1014 committed:
Merge branch 'master' into master

cctt1014 committed:
fix some bugs in command format and its exception #228 

fix some bugs in command format and its exception

therealnickcheong committed:
refactored logo into UI

therealnickcheong committed:
Merge remote-tracking branch 'upstream/master'

therealnickcheong committed:
refactored logo into UI

therealnickcheong committed:
Merge pull request #229 from therealnickcheong/master

refactored logo into UI

ChenChao19 committed:
Checkstyle

ChenChao19 committed:
Merge pull request #230 from ChenChao19/master

Checkstyle

therealnickcheong committed:
checkstyle and fixed goals test

therealnickcheong committed:
checkstyle money storage

Commit:
Merging team repo

Commit:
Merged team repo

therealnickcheong committed:
fixed instalment 2dp hopefully and find test update

therealnickcheong committed:
Merge pull request #231 from therealnickcheong/master

fixed checkstyle findtest and hopefully fixed 2dp for instalments

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W12-2/main

ChenChao19 committed:
fix 2dp issue

ChenChao19 committed:
Merge pull request #232 from ChenChao19/master

fix 2dp issue

Commit:
Commit to merge team repo

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W12-2/main

ChenChao19 committed:
updateDeleteinstalmentCommand

chengweixuan committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	dataFG/moneyAccount.txt

chengweixuan committed:
Fixed checkstyle for Ui and Goals

chengweixuan committed:
Fixed checkstyle for all JUnit of Archive, expenditure, income loans and shortcut timings

chengweixuan committed:
updated autocomplete

Commit:
testUndoAddIncome added

chengweixuan committed:
Merge pull request #233 from chengweixuan/master

Fixed checkstyle for Ui, parser, and all test code for Loan, Archive, Expenditure, Income and Timing Shortcuts

therealnickcheong committed:
fixed commit goal duplicate bug

therealnickcheong committed:
Merge pull request #234 from therealnickcheong/master

fixed commit goal duplicate bug

ChenChao19 committed:
Merge remote-tracking branch 'upstream/master'

ChenChao19 committed:
debug test

ChenChao19 committed:
checkstyleTest for Instalment and help package

ChenChao19 committed:
Merge pull request #235 from ChenChao19/master

checkstyleTest + debug for Test

Commit:
Undo JUnit tests for everything except loans and bank tracker

Commit:
Merged team repo

cctt1014 committed:
Merge pull request #30 from AY1920S1-CS2113T-W12-2/master

update the code

therealnickcheong committed:
fixed commit goal duplicate bug

therealnickcheong committed:
Merge pull request #236 from therealnickcheong/master

fixed test code checkstyle

cctt1014 committed:
add fix the checkstyle for C-statistcs

cctt1014 committed:
clear checkstyle errors for D-Bank

cctt1014 committed:
Fix checkstyle for C-statistics and D-bank #237 from cctt1014/master

Fix checkstyle for C-statistics and D-bank

cctt1014 committed:
Merge pull request #31 from AY1920S1-CS2113T-W12-2/master

update the code

cctt1014 committed:
fix the checkstyle in test file

Commit:
Added testUndoSettleLoans

Commit:
Merged with team repo

Chianhaoplanks committed:
Merge pull request #238 from Chianhaoplanks/master

Settled JUnit Tests for Undo

ChenChao19 committed:
making autocomplete faster

ChenChao19 committed:
Merge pull request #239 from ChenChao19/master

making autocomplete faster

cctt1014 committed:
fix change icon

cctt1014 committed:
fix bugs in graph command

cctt1014 committed:
add junit for graph command and javadoc for userIcon.java

cctt1014 committed:
Merge branch 'master' into master

cctt1014 committed:
Fix bugs in userIcon, add some junit and javadocs #240 from cctt1014/master

Fix bugs in user icon class, add some junit test and javadocs for graph and user icon

cctt1014 committed:
arrange all the gui controllers and launcher

cctt1014 committed:
arrange controllers into packages

cctt1014 committed:
Merge pull request #32 from AY1920S1-CS2113T-W12-2/master

update the code without rearrange the files

cctt1014 committed:
fix autocompleteTest

cctt1014 committed:
move the controllers into packages #241 from cctt1014/master

move the controllers into packages

therealnickcheong committed:
goals test fix commit goal upcomming

therealnickcheong committed:
Merge pull request #242 from therealnickcheong/master

goals test fix commit goal upcomming

therealnickcheong committed:
fix done commit and delete goal bugs

therealnickcheong committed:
Merge remote-tracking branch 'upstream/master'

cctt1014 committed:
move the list of bank trackers to the right hand side pane

cctt1014 committed:
move the list of bank trackers to the right hand side pane #243 from cctt1014/master

move the list of bank trackers to the right hand side pane

therealnickcheong committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	dataFG/moneyAccount.txt

therealnickcheong committed:
Merge pull request #244 from therealnickcheong/master

bugs fixed for goals done delete and commit

chengweixuan committed:
Fixed negative add commands error

therealnickcheong committed:
checkstyle mainwindow add goal

cctt1014 committed:
Merge pull request #33 from AY1920S1-CS2113T-W12-2/master

update the code

Commit:
Done JUnit Undo Commands

Chianhaoplanks committed:
Merge pull request #245 from Chianhaoplanks/master

Done JUnit Undo Commands

chengweixuan committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	data/account-test.txt
#	dataFG/moneyAccount.txt
#	src/test/java/money/LoansTest.java

Commit:
Added JavaDoc comments for MoneyStorage

Chianhaoplanks committed:
Merge pull request #246 from Chianhaoplanks/master

Added JavaDoc comments for MoneyStorage

chengweixuan committed:
Fixed negative add commands error for instalment and loans

cctt1014 committed:
add junit for list bank trackers

chengweixuan committed:
fixed loan listing inconsistency

chengweixuan committed:
Merge remote-tracking branch 'upstream/master'

chengweixuan committed:
Merge pull request #247 from chengweixuan/master

Fixed negative amount values for add commands for Income, Expenditure, Instalment and Loans. Fixed other bugs

therealnickcheong committed:
Merge remote-tracking branch 'upstream/master'

therealnickcheong committed:
fixed fat L jar bug

therealnickcheong committed:
Merge pull request #248 from therealnickcheong/master

 fixed the fat L bugs

cctt1014 committed:
add exception for create new bank tracker

cctt1014 committed:
Merge branch 'master' into master

cctt1014 committed:
Merge pull request #34 from AY1920S1-CS2113T-W12-2/master

update the code

therealnickcheong committed:
fixed fat change icon L jar bug

therealnickcheong committed:
Merge pull request #249 from therealnickcheong/master

fixed fat change icon L jar bug

cctt1014 committed:
add exception for the wrong format in InternalTransferCommand

cctt1014 committed:
add exception of the command format for D-Bank #250 from cctt1014/master

add exception of the command format for D-Bank

cctt1014 committed:
Merge pull request #35 from AY1920S1-CS2113T-W12-2/master

update the code

therealnickcheong committed:
fixed fat build.gradle L bug

cctt1014 committed:
fix change icon

therealnickcheong committed:
Merge pull request #251 from therealnickcheong/master

fixed fat build.gradle L bug

cctt1014 committed:
Merge pull request #36 from AY1920S1-CS2113T-W12-2/master

update the code

cctt1014 committed:
fix change icon #252 from cctt1014/master

fix change icon

chengweixuan committed:
Updated README and aboutUS

chengweixuan committed:
Fixed Loan JUnit changes due to exception handling

chengweixuan committed:
Updated table of contents for README

chengweixuan committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	build.gradle
#	data/account-test.txt

chengweixuan committed:
Merge pull request #253 from chengweixuan/master

Updated ReadME file and AboutUs file. Linked Dev guide and User guide to readme. Fixed JUnit due to changes in exception

chengweixuan committed:
fixed toc

chengweixuan committed:
fixed toc heading

chengweixuan committed:
Merge remote-tracking branch 'upstream/master'

chengweixuan committed:
Merge pull request #254 from chengweixuan/master

Fixed table of contents for README

chengweixuan committed:
Update README.adoc

chengweixuan committed:
Fixed TOC

chengweixuan committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	README.adoc

chengweixuan committed:
Merge pull request #255 from chengweixuan/master

Update linking for TOB in README

chengweixuan committed:
added about us to READme

chengweixuan committed:
Merge pull request #256 from chengweixuan/master

Added aboutUs to readme

chengweixuan committed:
Update README.adoc

therealnickcheong committed:
fixed goal test bug that arose from bug fixing

therealnickcheong committed:
Merge pull request #257 from therealnickcheong/master

fixed goal test bug that arose from bug fixing

therealnickcheong committed:
all checkstyle fixed

therealnickcheong committed:
Merge pull request #258 from therealnickcheong/master

all checkstyle fixed

therealnickcheong committed:
all checkstyle fixed

therealnickcheong committed:
Merge pull request #259 from therealnickcheong/master

refactored duke to FG

cctt1014 committed:
Merge pull request #37 from AY1920S1-CS2113T-W12-2/master

update the code

cctt1014 committed:
add exception for check future balance

cctt1014 committed:
add exception for check future balance #260 from cctt1014/master

add an exception for check future balance

chengweixuan committed:
Added exception to only allow unique loan descriptions

chengweixuan committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	README.adoc
#	dataFG/moneyAccount.txt

chengweixuan committed:
Merge pull request #261 from chengweixuan/master

Added exception to allow only Loans with unique descriptions

cctt1014 committed:
add exception for graph command handling wrong graph type

cctt1014 committed:
Merge pull request #38 from AY1920S1-CS2113T-W12-2/master

update the code

cctt1014 committed:
add an exception for graph command handling wrong graph type #262 from cctt1014/master

add an exception for graph command handling wrong graph type

cctt1014 committed:
fix graph command history

cctt1014 committed:
Merge branch 'master' of https://github.com/cctt1014/main

cctt1014 committed:
add authorship

cctt1014 committed:
Merge pull request #263 from cctt1014/master

fix graph command history

chengweixuan committed:
Delete UserGuide.pdf

chengweixuan committed:
Added updated UG

chengweixuan committed:
Update AboutUs.adoc

ChenChao19 committed:
Last

ChenChao19 committed:
last last

ChenChao19 committed:
really last

ChenChao19 committed:
Merge pull request #264 from ChenChao19/master

Last

chengweixuan committed:
Delete DeveloperGuide.pdf

chengweixuan committed:
Added updated Dev Guide

therealnickcheong committed:
FINAL COMMIT

therealnickcheong committed:
Merge pull request #265 from therealnickcheong/master

FINAL COMMIT

ChenChao19 committed:
Add files via upload

ChenChao19 committed:
Create PPP

ChenChao19 committed:
Delete PPP

ChenChao19 committed:
Delete [CS2113T-W12-2][Chen Chao]PPP.pdf.docx

ChenChao19 committed:
Create PPP

ChenChao19 committed:
Delete PPP

ChenChao19 committed:
Add files via upload

ChenChao19 committed:
Delete [CS2113T-W12-2][Chen Chao]PPP.pdf.docx

ChenChao19 committed:
Add files via upload
